nano/node/repcrawler.hpp

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <boost/multi_index/hashed_index.hpp>
+#include <boost/multi_index/member.hpp>
+#include <boost/multi_index/ordered_index.hpp>
+#include <boost/multi_index/random_access_index.hpp>
+#include <boost/multi_index_container.hpp>
+#include <chrono>
+#include <memory>
+#include <nano/node/common.hpp>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace mi = boost::multi_index;
+
+namespace nano
+{
+class node;
+// clang-format off
+class tag_endpoint {};
+class tag_last_request {};
+class tag_account {};
+class tag_weight {};
+// clang-format on
+
+/**
+ * A representative picked up during repcrawl.
+ */
+class representative
+{
+public:
+	representative (nano::account account_a, nano::amount weight_a, nano::endpoint endpoint_a) :
+	account (account_a), weight (weight_a), endpoint (endpoint_a)
+	{
+	}
+	nano::account account{ 0 };
+	nano::amount weight{ 0 };
+	nano::endpoint endpoint;
+	std::chrono::steady_clock::time_point last_request{ std::chrono::steady_clock::time_point () };
+	std::chrono::steady_clock::time_point last_response{ std::chrono::steady_clock::time_point () };
+};
+
+using PROBABLE_REP_TYPE = boost::multi_index_container<
```

Comment:
It looks like this container is only accessed within rep_crawler now which is an improvement in encapsulation. Can we move this type and the tags privately inside the rep_crawler class so they aren't in the nano namespace.

Reply:
All uppercase here looks like a constant.  probably_rep_t seems good.

Reply:
Good points, will fix.